jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Attributes.java

```diff
@@ -67,18 +67,25 @@
     // TODO something better than this
     default Map<String, Object> asAttributeMap()
     {
+        final Set<String> attributeNameSet = getAttributeNameSet();
+
         return new AbstractMap<>()
         {
+            @Override
+            public int size()
+            {
+                return attributeNameSet.size();
+            }
+
             @Override
             public Set<Entry<String, Object>> entrySet()
             {
                 return new AbstractSet<>()
                 {
-                    Iterator<String> names = getAttributeNameSet().iterator();
-
                     @Override
                     public Iterator<Entry<String, Object>> iterator()
                     {
+                        Iterator<String> names = attributeNameSet.iterator();
```

Comment:
Down below, you do: `new Map.Entry<>() ...`.
Do `Map.Entry.of(name, getAttribute(name))` as it's much cleaner.